packages/@aws-cdk/aws-apigatewayv2/lib/common/domain-name.ts

```diff
@@ -139,4 +195,35 @@ export class DomainName extends Resource implements IDomainName {
       truststoreVersion: mtlsConfig.version,
     };
   }
+
+  /**
+   * Adds a configuration to a domain name. Properties like certificate, endpoint type and security policy can be set using this method.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigatewayv2-domainname-domainnameconfiguration.html
+   * @param props domain name properties to be set
+   */
+  public addEndpoint(props: DomainNameProps) : void {
+    const domainNameConfig: CfnDomainName.DomainNameConfigurationProperty = {
+      certificateArn: props.certificate.certificateArn,
+      certificateName: props.certificateName,
+      endpointType: props.endpointType ? props.endpointType?.toString() : 'REGIONAL',
+      ownershipVerificationCertificateArn: props.ownershipCertificate?.certificateArn,
+      securityPolicy: props.securityPolicy?.toString(),
+    };
+
+    if (this.isDuplicateEndpointType(domainNameConfig.endpointType)) {
+      throw new Error('no two domain name configurations should have the same endpointType');
+    }
+
+    this.domainNameConfigurations.push(domainNameConfig);
+  }
+
+  // validates whether the new domain name configuration has a unique endpoint or not
+  private isDuplicateEndpointType(endpointType: string | undefined) : boolean {
+    for (let config of this.domainNameConfigurations) {
+      if (endpointType && endpointType == config.endpointType) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
If you could change it such that the error is thrown from this method directly, instead of returning a bool, the error message can say "an endpoint with type ${endpointType} already exists".

The current message doesn't state what type it is which can be improved.